### PR TITLE
chore(agent): require min length for code shown to complete

### LINKED ISSUE #2953: Having min length for code shown to complete (e.g ignore completions <=3 characters)
Example: https://github.com/TabbyML/tabby/blob/main/clients/tabby-agent/src/postprocess/dropBlank.ts

### PR DESCRIPTION
**Pull request by shankarg87**
fix #2953 

caveat: new to typescript

### CODE REVIEW

File: clients/tabby-agent/src/postprocess/index.ts [hunk 1]
```diff
@@ -53,6 +53,6 @@ export async function postCacheProcess(
     .then(applyFilter(formatIndentation))
     .then(applyFilter(dropDuplicated))
     .then(applyFilter(trimSpace))
-    .then(applyFilter(dropBlank))
+    .then(applyFilter(dropMinimum))
```
Comment (wsxiaoys) [inline clients/tabby-agent/src/postprocess/index.ts:56]:
```suggestion
    .then(applyFilter(dropBlank))
    .then(applyFilter(dropMinimum))
```

File: clients/tabby-agent/src/postprocess/base.ts [hunk 2]
```diff
@@ -1,6 +1,9 @@
 import { CompletionItem } from "../CompletionSolution";
 import { getLogger } from "../logger";
+import { AgentConfig } from "../AgentConfig";
 
-export type PostprocessFilterFactory = (config: unknown) => PostprocessFilter;
+export type PostprocessFilterFactory =
+  | (() => PostprocessFilter)
+  | ((config: AgentConfig["postprocess"]) => PostprocessFilter);
```
Comment (wsxiaoys) [inline clients/tabby-agent/src/postprocess/base.ts:7]: should this be (config: AgentConfig)?

cc @icycodes

**Comment by shankarg87** [comment]
anyway to trigger a rerun of failed tests (without additional push to PR?)

**Comment by wsxiaoys** [comment]
> anyway to trigger a rerun of failed tests (without additional push to PR?)

you shall be able to run mocha test locally, fixing it and push commit for a new run

File: clients/tabby-agent/src/postprocess/base.ts (hunk 2 above)
Comment (wsxiaoys) [reply clients/tabby-agent/src/postprocess/base.ts:7]: that's possible, but just leave @icycodes to double confirm

File: clients/tabby-agent/src/postprocess/index.ts [hunk 3]
```diff
@@ -53,6 +53,6 @@ export async function postCacheProcess(
     .then(applyFilter(formatIndentation))
     .then(applyFilter(dropDuplicated))
     .then(applyFilter(trimSpace))
-    .then(applyFilter(dropBlank))
```
Comment (wsxiaoys) [inline clients/tabby-agent/src/postprocess/index.ts:56]: guess the code of dropBlank can be removed as well?

**Review by wsxiaoys — CHANGES_REQUESTED** [review]

File: clients/tabby-agent/src/AgentConfig.ts [hunk 4]
```diff
@@ -71,6 +71,7 @@ export type AgentConfig = {
   postprocess: {
     limitScope: any;
     calculateReplaceRange: any;
+    minChars: number;
```
Comment (wsxiaoys) [inline clients/tabby-agent/src/AgentConfig.ts:74]:
```suggestion
    minCompletionChars: number;
```

File: clients/tabby-agent/src/AgentConfig.ts [hunk 5]
```diff
@@ -172,6 +173,7 @@ export const defaultAgentConfig: AgentConfig = {
   postprocess: {
     limitScope: {},
     calculateReplaceRange: {},
+    minChars: 4,
```
Comment (wsxiaoys) [inline clients/tabby-agent/src/AgentConfig.ts:176]: ditto

**Review by wsxiaoys — APPROVED** [review]
Otherwise LGTM

File: clients/tabby-agent/src/postprocess/index.ts (hunk 3 above)
Comment (shankarg87) [reply clients/tabby-agent/src/postprocess/index.ts:56]: done, ptal

**Review by icycodes — APPROVED** [review]

**Pull request merged**